### ISSUE TITLE
Vue.js as SPA?

### ISSUE THREAD

**Issue by illumen** [NeutroniumCore/Neutronium#26]
Hi David.

First let me say you've done a great job developing and maintaining this project.

Now, how would one best implement a SPA solution with Neutronium?
It feels a bit illogical to have a separate VueJS project for each view instead of one vue project to handle all views especially since vue can handle routing perfectly well.

However, one would not want to forego the ViewModels aswell, so is there a way to implement a single vue-project where perhaps vue handles routing but triggers Neutronium to swap ViewModel?

Or maybe you see something else as a better solution?

**Comment by David-Desmaisons** [comment]
I will go this way:

**On C# side:**
```
MainWindow
|_____PageViewModel
```

Where `MainWindow` will contain all the comands and information that will stay relevant during the whole session.
`PageViewModel` typically will an interface ( `IPageViewModel `) and will contain all data that are relative to the current page.

**On vue side:**
Then you can use [dynamic component](https://vuejs.org/v2/guide/components.html#Dynamic-Components) to display the current page.

For example `IPageViewModel `may have `component` property that will reference the vue component responsible for displaying the component:
```C#
public interface IPageViewModel 
{
   string ComponentName {get;}
}
```

```HTML
<component :is="PageViewModel.ComponentName" :vm="PageViewModel">
</component>
```

**Comment by David-Desmaisons** [comment]
@illumen Did this provide the answer you were looking for?

**Comment by genyx** [comment]
When i overwrite the `DataContext` Property with the PageViewModel, the Browser refreshs for an instant. When i try only to have variable data in an say `Data` Object, this Class is not getting updated. Even tho i implemented `INotifyPropertyChanged` and called `OnPropertyChanged()`
Which is the best way to refresh data when "opening a new page" ?

**Comment by David-Desmaisons** [comment]
@genyx, you should open a separated issue sharing the implementation of the viewmodel so I can investigate.

**Comment by David-Desmaisons** [comment]
@genyx , I think I guess what the problem is. I recently correct a bug where in some scenarios INotifyPropertyChanged  triggerred during update from javascript where not decteted. As a tempory bypass, you can try to add  `await Task.Yield()` before sending the `INotifyPropertyChanged` event linked to the `Data` property,
This bus is fixed in upcoming version 1.0.0.

**Comment by genyx** [comment]
I think there was a missunderstanding on my side. I have not yet managed to implement the variable binding stuff for "sub"-components. I have to pass them separatly with javascript, am i right?

**Comment by David-Desmaisons** [comment]
@genyx I will set-up a demo-application using this technique (and maybe alternative with vue-router) after releasing version v.1.0,0

**Comment by David-Desmaisons** [comment]
@genyx I just released version 1.0.0

**Comment by David-Desmaisons** [comment]
See [Neutronium.SPA.Demo](https://github.com/NeutroniumCore/Neutronium.SPA.Demo) as a reference SPA implementation.
